### ISSUE TITLE
Usage of non-standard naming convention for enumerations

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/go-sourcegraph#83]
As discovered in https://github.com/sourcegraph/go-sourcegraph/pull/75#discussion_r43541743 we use a non-standard naming convention for enumerations (per [the official protobuf style guide](https://developers.google.com/protocol-buffers/docs/style#enums)).

We should switch from the Go style of `CamelCase` to the official protobuf style of `UPPER_CASE` instead, and ensure that `protobuf-gen-gogo` or whatever generates the Go source is emitting these with nice Go names as we desire.

This is important for consistency esp. with regard to supporting other languages.

**Comment by shurcooL** [comment]
I agree. Our Go code should follow the Go style, and protobuf code should use protobuf style.

That's the right thing to do, even though it means following/knowing 2 styles for us.

But it's going to be friendlier to people using other languages or more familiar with protobufs.
